Fill.py

```diff
@@ -167,31 +170,41 @@ def sweep_from_pool():
         return new_state
 
     while itempool and locations:
-        item_to_place = itempool.pop()
+        items_to_place = []
+        nextpool = []
+        placing_players = set()
+        for item in reversed(itempool):
+            if item.player not in placing_players:
+                placing_players.add(item.player)
+                items_to_place.append(item)
+            else:
+                nextpool.insert(0, item)
+        itempool = nextpool
```

Comment:
Not clear to me if this is really needed. The claimed reason for this loop does not seem to apply anymore.